Documentation/CRDs/Cluster/external-cluster/topology-for-external-mode.md

```diff
@@ -0,0 +1,199 @@
+
+The import script will then create a new storage class named `ceph-rbd-topology`.
+
+## Example Configuration
+
+### Ceph cluster
+
+Looking at the osd tree, before jumping to the steps
+```
+ID  CLASS  WEIGHT   TYPE NAME       STATUS  REWEIGHT  PRI-AFF
+-1         2.00000  root default
+-9         0.50000      host osd-0
+ 3    hdd  0.25000          osd.3       up   1.00000  1.00000
+ 7    hdd  0.25000          osd.7       up   1.00000  1.00000
+-7         0.50000      host osd-1
+ 1    hdd  0.25000          osd.1       up   1.00000  1.00000
+ 6    hdd  0.25000          osd.6       up   1.00000  1.00000
+-3         0.50000      host osd-2
+ 2    hdd  0.25000          osd.2       up   1.00000  1.00000
+ 5    hdd  0.25000          osd.5       up   1.00000  1.00000
+-5         0.50000      host osd-3
+ 0    hdd  0.25000          osd.0       up   1.00000  1.00000
+ 4    hdd  0.25000          osd.4       up   1.00000  1.00000
+```
+
+Create m replica-n pools, in the example create 3 replica-1 pools:
+
+```
+pool 10 'ab' replicated size 1 min_size 1 crush_rule 5 object_hash rjenkins pg_num 32 pgp_num 32 autoscale_mode on last_change 334 lfor 0/0/254 flags hashpspool,selfmanaged_snaps stripe_width 0 application rbd read_balance_score 1.06
+pool 11 'cd' replicated size 1 min_size 1 crush_rule 6 object_hash rjenkins pg_num 32 pgp_num 32 autoscale_mode on last_change 335 lfor 0/0/256 flags hashpspool,selfmanaged_snaps stripe_width 0 application rbd read_balance_score 1.19
+pool 12 'ef' replicated size 1 min_size 1 crush_rule 7 object_hash rjenkins pg_num 32 pgp_num 32 autoscale_mode on last_change 336 lfor 0/0/256 flags hashpspool,selfmanaged_snaps stripe_width 0 application rbd read_balance_score 1.06
+```
+
+!!! note
+   Require to run `ceph config set global  mon_allow_pool_size_one true` for creating replica-1 pools
+
+Review the specific crush rules:
+crush_rule 5
+```
+ {
+            "rule_id": 5,
+            "rule_name": "rule_host-osd0-hdd",
+            "type": 1,
+            "steps": [
+                {
+                    "op": "take",
+                    "item": -10,
+                    "item_name": "osd-0~hdd"
+                },
+                {
+                    "op": "choose_firstn",
+                    "num": 0,
+                    "type": "osd"
+                },
+                {
+                    "op": "emit"
+                }
+            ]
+}
+```
+
+Once the pools are ready with specific crush rules.
+
+Decide what will be the flags value based on the crush rules and kubernetes cluster nodes.
+
+--topology-pools -> Would be the 3 replica-1 pools, i.e ab,cd,ef
+--topology-failure-domain-label -> kubernetes cluster nodes label depending on the need.
+--topology-failure-domain-values -> kubernetes cluster nodes
+
+Current machine:
+```
+$ kubectl get nodes --show-labels
```

Comment:
No need to show the list of nodes. Maybe just tell them a command to query the nodes in a zone: `kubectl get node -l topology.kubernetes.io/zone=zone-a`